config/esp32/components/chip/Kconfig

```diff
@@ -163,6 +163,13 @@ menu "CHIP Core"
             help
                 Option to enable/disable CHIP data model.
 
+        config CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
+            bool "Use heap memory for system buffer pool"
+            default y
+            help
+                Option to use heap memory for system buffer pool.
```

Comment:
What is "system buffer pool"?

Setting CHIP_SYSTEM_CONFIG_POOL_USE_HEAP will make all ObjectPool instances:

1. Use heap.
2. Not actually enforce any size limits (is that what you want?)

It does not affect the behavior of SystemPacketBuffer, which is the first thing "system buffer" made me think of....

Reply:
Sorry for the misunderstanding about what this option does. I have updated the prompt and help text.

This came up when we started seeing DRAM overflow in the all-clusters-app/esp32. On enabling this it adds about 19K of DRAM.

Setting this default to `n`. Can be used by whoever needs the DRAM but okay to allocate infinite number of object pools.